idea-plugin/src/main/kotlin/arrow/meta/ide/plugins/proofs/markers/propertyCoercion.kt

```diff
@@ -0,0 +1,43 @@
+package arrow.meta.ide.plugins.proofs.markers
+
+import arrow.meta.ide.IdeMetaPlugin
+import arrow.meta.ide.plugins.proofs.inspections.PairTypes
+import arrow.meta.ide.plugins.proofs.inspections.PairTypes.Companion.pairOrNull
+import arrow.meta.ide.plugins.proofs.inspections.resolveKotlinType
+import arrow.meta.ide.resources.ArrowIcons
+import arrow.meta.phases.CompilerContext
+import arrow.meta.phases.ExtensionPhase
+import arrow.meta.plugins.proofs.phases.areTypesCoerced
+import arrow.meta.plugins.proofs.phases.coerceProof
+import org.jetbrains.kotlin.nj2k.postProcessing.type
+import org.jetbrains.kotlin.psi.KtProperty
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
+
+val IdeMetaPlugin.implicitCoercionPropertyLineMarker: ExtensionPhase
+  get() = addLineMarkerProvider(
+    icon = ArrowIcons.ICON4,
+    composite = KtProperty::class.java,
+    transform = { psiElement ->
+      psiElement.ctx()?.let { ctx ->
+        psiElement.safeAs<KtProperty>()?.takeIf {
```

Comment:
Aren't we sure it's a `KtProperty` already given the `composite` param?

Reply:
yes, but its preferable to use `safeAs` than casting it like.. psi as KtProperty

Reply:
I see. I would expect the DSL to provide you the already casted `KtProperty` in the `transform` block since you already know that is what it is, but not sure if there's a reason to not do that. @i-walker ?

Reply:
this part of the DSL has other intentions. 
transform is always the entry point for targeting the elements the user care about. The composite is only there so one can use it for something like `message` or other complementary parameters. Otherwise, the user has to know how to do that.